codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustWriter.kt

```diff
@@ -269,6 +271,20 @@ fun <T : AbstractCodeWriter<T>> T.docs(text: String, vararg args: Any, newlinePr
     return this
 }
 
+/**
+ * Generates a `#[deprecated]` attribute for [shape].
+ */
+fun RustWriter.deprecatedShape(shape: Shape): RustWriter {
+    val deprecatedTrait = shape.getTrait<DeprecatedTrait>() ?: return this
+
+    val note = deprecatedTrait.message.orElse("")
+    val since = deprecatedTrait.since.orElse("")
```

Comment:
See the comment about optional strings in the file above.

Reply:
I cannot find any comment about that. Could you provide a link to those lines?

Reply:
Or you can use `let` with a nullable. I believe @crisidev refers to https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
I am referring to Kotlin nullability: https://kotlinlang.org/docs/null-safety.html#checking-for-null-in-conditions